### PR TITLE
Update WebPush Console to work with node-webpush-server

### PR DESCRIPTION
**Pull request by danbev**
To review and test please follow the instructions in [client-netty/README.md](https://github.com/danbev/aerogear-webpush-server/tree/console-with-nodejs/client-netty#using-the-webpush-console)

Jira: https://issues.jboss.org/browse/AGPUSH-1271

### CODE REVIEW

**Comment by danbev** [comment]
@lholmquist @matzew Would you guys mind doing a review and test of this?

**Comment by matzew** [comment]
>Java 1.8 is required for the AeroGear WebPush Server.
```
cd server-netty
mvn exec:exec
```

I am getting this:
```
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (default-cli) on project aerogear-webpush-server-netty: Command execution failed. Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Comment by matzew** [comment]
@danbev do you have the `demo.key` and `selfsigned.crt` somewhere? Or any instructions on how to actually create those ?

**Comment by danbev** [comment]
@matzew Links to ```demo.key``` and ```selfsigned.cert``` can be found in the readme:
https://github.com/danbev/aerogear-webpush-server/tree/console-with-nodejs/client-netty#start-the-nodejs-server

**Comment by danbev** [comment]
@matzew For you first error I think I need to add instructions to build it first. Or perhaps like that goal with the compile goal or whatever maven calls it.

**Comment by matzew** [comment]
> Links to demo.key and selfsigned.cert can be found in the readme:

@danbev ah, now I see them. Not sure the note is easy to spot :-)

**Comment by danbev** [comment]
@matzew How about if we make them bold/highlighted as in ```demo.key```, would that be enough?

**Comment by matzew** [comment]
@danbev on the node server:

First did the `npm install -g webpush-server` as suggested on the repo,

now running `webpush-server --port 8080 --key demo.key --cert selfsigned.crt`

But I am getting:
```
➜  WEB_PUSH  webpush-server --port 8080 --key demo.key --cert selfsigned.crt
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: listen EADDRINUSE
    at exports._errnoException (util.js:746:11)
    at Server._listen2 (net.js:1129:14)
    at listen (net.js:1155:10)
    at net.js:1253:9
    at dns.js:85:18
    at process._tickDomainCallback (node.js:381:11)
    at Function.Module.runMain (module.js:503:11)
    at startup (node.js:129:16)
    at node.js:814:3
``

here is my node version:
➜  WEB_PUSH  node -v
v0.12.0

**Comment by matzew** [comment]
perhaps "pathToNettyResources/demo.ky" ?

On Wed, Feb 11, 2015 at 3:19 PM, Daniel Bevenius <notifications@github.com>
wrote:

> @matzew <https://github.com/matzew> How about if we make them
> bold/highlighted as in demo.key, would that be enough?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-webpush-server/pull/4#issuecomment-73887601>
> .
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf

**Comment by danbev** [comment]
@matzew Looks like you have something running on port ```8080``` already.

**Comment by matzew** [comment]
Ah, thanks for the good "listen EADDRINUSE" error :-)

On Wed, Feb 11, 2015 at 3:28 PM, Daniel Bevenius <notifications@github.com>
wrote:

> @matzew <https://github.com/matzew> Looks like you have something running
> on port 8080 already.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-webpush-server/pull/4#issuecomment-73889044>
> .
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf

File: client-netty/README.md [hunk 1]
```diff
@@ -1,62 +1,156 @@
...
+
+    $ webpush-server --port 8080 --key demo.key --cert selfsigned.crt
+    Listening on https://Daniels-MBP.lan:8080...
+
+The _demo.key_ and _selfsigned.crt_ can be found in [server-netty/src/main/resources](../server-netty/src/main/resources).
+
 ### Start the console
-Press ```tab``` to see the available commands and also to display the available options for specific commands.   
+__Java 1.8__ is required for the WebPush Console when connecting to the AeroGear WebPush Server, and __Java 1.7__ for
+the node-webpush-server.
```
Comment (matzew) [inline client-netty/README.md]: @danbev How about:

**Java 1.8** is required for the WebPush Console when connecting to the _AeroGear WebPush Server_.
**Java 1.7** is required for the WebPush Console when connecting to the for the _node-webpush-server_.

**Comment by danbev** [comment]
@matzew I've updated the PR now. Let me know if that makes it clearer.

**Comment by matzew** [comment]
Trying the client connect:

```
Exception in thread "AeshProcess: 6" java.lang.NoClassDefFoundError: Could not initialize class io.netty.handler.ssl.JdkNpnApplicationProtocolNegotiator
	at io.netty.handler.ssl.JdkSslContext.toNegotiator(JdkSslContext.java:268)
	at io.netty.handler.ssl.JdkSslClientContext.<init>(JdkSslClientContext.java:161)
	at io.netty.handler.ssl.SslContext.newClientContext(SslContext.java:483)
	at io.netty.handler.ssl.SslContext.newClientContext(SslContext.java:434)
	at org.jboss.aerogear.webpush.WebPushClient.configureSsl(WebPushClient.java:189)
	at org.jboss.aerogear.webpush.WebPushClient.connect(WebPushClient.java:88)
	at org.jboss.aerogear.webpush.WebPushConsole$ConnectCommand.execute(WebPushConsole.java:110)
	at org.jboss.aesh.console.AeshConsoleImpl$AeshConsoleCallbackImpl.execute(AeshConsoleImpl.java:325)
	at org.jboss.aesh.console.AeshProcess.run(AeshProcess.java:40)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by matzew** [comment]
On the CONNECT, the README says:

```
[webpush]$ connect -h hostname -p 8443
Connected to [hostname:8443]
```

should that state `-p port`, since the node-server runs on 8080, while the Java server runs on 8443?

**Comment by danbev** [comment]
@matzew When you got the error above, what Java 7 version were you using? 
Yeah, that makes sense to change ```-p port```. I'll fix that.

**Comment by matzew** [comment]
java version "1.7.0_65"
Java(TM) SE Runtime Environment (build 1.7.0_65-b17)
Java HotSpot(TM) 64-Bit Server VM (build 24.65-b04, mixed mode)

On Wed, Feb 11, 2015 at 3:52 PM, Daniel Bevenius <notifications@github.com>
wrote:

> @matzew <https://github.com/matzew> When you got the error above, what
> Java 7 version were you using?
> Yeah, that makes sense to change -p port. I'll fix that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-webpush-server/pull/4#issuecomment-73893147>
> .
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf

**Comment by danbev** [comment]
@matzew Thanks, let me try that version and see what the issue is.

**Comment by danbev** [comment]
@matzew I managed to install 1.7.0_65:
```
$ java -version
java version "1.7.0_65"
Java(TM) SE Runtime Environment (build 1.7.0_65-b17)
Java HotSpot(TM) 64-Bit Server VM (build 24.65-b04, mixed mode)
```
But I can't reproduce the issue you are seeing. Could you paste the output of the following command?
```mvn compile exec:exec -X```

**Comment by matzew** [comment]
Caused by: java.lang.RuntimeException: NPN unsupported.

hrm..... I get this from `mvn -v` -> Java version: 1.7.0_65, vendor: Oracle
Corporation

On Wed, Feb 11, 2015 at 4:49 PM, Daniel Bevenius <notifications@github.com>
wrote:

> @matzew <https://github.com/matzew> I managed to install 1.7.0_65:
>
> $ java -version
> java version "1.7.0_65"
> Java(TM) SE Runtime Environment (build 1.7.0_65-b17)
> Java HotSpot(TM) 64-Bit Server VM (build 24.65-b04, mixed mode)
>
> But I can't reproduce the issue you are seeing. Could you paste the output
> of the following command?
> mvn compile exec:exec -X
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-webpush-server/pull/4#issuecomment-73903795>
> .
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf

**Comment by danbev** [comment]
This is the same command on my machine:
```
$ mvn -v
Apache Maven 3.2.1 (ea8b2b07643dbb1b84b6d16e1f08391b666bc1e9; 2014-02-14T18:37:52+01:00)
Maven home: /Users/danielbevenius/work/tools/maven/apache-maven-3.2.1
Java version: 1.7.0_65, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.7.0_65.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.10.2", arch: "x86_64", family: "mac"
```

Could you paste the output of ```mvn compile exec:exec -X```? I'd like to see what the bootclasspath looks like.

**Comment by lholmquist** [comment]
i'll be testing in a few

**Comment by matzew** [comment]
@danbev here is a longer gist:

https://gist.github.com/matzew/043c2fe10f0ad5470dc1

**Comment by lholmquist** [comment]
walking through the steps, i had no issues with maven or anything.  Was able to connect, register, monitor and subscribe as well as get a message

i'm on 10.10.2,   `java version "1.7.0_40"` and `java version "1.8.0_25"`,  node version `0.12.0`

The one thing i did need to do(or not do),  was to not include the `https://` when connecting

**Comment by danbev** [comment]
@lholmquist Thanks! Regarding the ```https://```, one should only specify the hostname that is displayed when starting the node-webpush-server. Sounds like that is not clear enough. Let me try to clarify that

**Comment by abstractj** [comment]
Hi there, after the latest changes I got it working here and manage to run the console. Thanks @danbev for your patience

+1

**Comment by matzew** [comment]
@danbev late here, but a complete start over made it work on my side - not sure what I missed before.

**Comment by danbev** [comment]
@matzew I added a note about there reason for you failure you were seeing here:
https://gist.github.com/matzew/043c2fe10f0ad5470dc1#comment-1392670. 
Sorry about this, I should have thought of this sooner.

**Pull request merged**